### PR TITLE
Rewritten external identity storage, added support for Redis

### PR DESCRIPTION
**Pull request by rogeralsing**

### CODE REVIEW

File: benchmarks/ClusterBenchmark/Program.cs [hunk 1]
```diff
@@ -159,10 +160,11 @@ private static IClusterProvider ClusterProvider()
             }
         }
 
-        private static MongoIdentityLookup GetIdentityLookup()
+        private static IIdentityLookup GetIdentityLookup()
         {
             var db = GetMongo();
-            var identity = new MongoIdentityLookup("mycluster", db);
+            var identity = new IdentityStorageLookup(
+                new MongoIdentityStorage("mycluster", db.GetCollection<PidLookupEntity>("pids")));
```
Comment (rogeralsing) [inline benchmarks/ClusterBenchmark/Program.cs:167]: Should we make "pids" configurable?

Comment (mhelleborg) [reply benchmarks/ClusterBenchmark/Program.cs:167]: This is configurable, since it is passed in by the user

Comment (rogeralsing) [reply benchmarks/ClusterBenchmark/Program.cs:167]: doh, it's in the consumer code, my bad

**Comment by rogeralsing** [comment]
Looks rock solid, I can't find anything that needs changing. 👍🏻

**Comment by mhelleborg** [comment]
I have one more change I want to add, but it requires changing the IIdentityLookup interface. When retrieving a PID from IdentityLookup and getting a dead letter response, we should purge it..

**Comment by mhelleborg** [comment]
@rogeralsing Should we update the IdentityLookup interface to support this?

**Comment by rogeralsing** [comment]
Sure lets go for that

**Pull request merged**